test/common/stats/stats_impl_test.cc

```diff
@@ -466,5 +468,35 @@ TEST(TagProducerTest, CheckConstructor) {
       "No regex specified for tag specifier and no default regex for name: 'test_extractor'");
 }
 
+TEST(SomeTest, HotAllocConsistencyTest) {
+  int max_stats_ = 16384; // ENVOY_DEFAULT_MAX_STATS
```

Comment:
If you want this to match that number, you should probably use the macro directly.

Reply:
I'll include `server/options_impl.h` and get access to it.